rumqttc/src/tls.rs

```diff
@@ -102,31 +107,31 @@ pub async fn rustls_connector(tls_config: &TlsConfiguration) -> Result<RustlsCon
 
             // Add der encoded client cert and key
             let mut config = if let Some(client) = client_auth.as_ref() {
+
                 let certs =
                     rustls_pemfile::certs(&mut BufReader::new(Cursor::new(client.0.clone())))?;
-                // load appropriate Key as per the user request. The underlying signature algorithm
+                if certs.is_empty() {
+                    return Err(Error::NoValidCertInChain);
+                }
+                // Load appropriate Key as per the user request. The underlying signature algorithm
```

Comment:
outdated comment, we no longer are loading `Key` as per user request. Rather we are extracting first valid key from file provided by user.

Reply:
Removed